src/effects/EffectUI.cpp

```diff
@@ -508,6 +508,13 @@ void EffectUIHost::OnClose(wxCloseEvent & WXUNUSED(evt))
    DoCancel();
    CleanupRealtime();
 
+   // Flush settings when the dialog closes.  But please note -- this makes
+   // close always commit settings changes.  If it is decided to distinguish
+   // a cancellation of the dialog from a commit, then this call should move
+   // somewhere else.
+   if (mpAccess)
```

Comment:
Should we distinguish between realtime and destructive processing here? Probably we should.

Reply:
I think not.  If you step through it, you will see that for destructive application, an EffectSettingsAccessTee flushes to a SimpleEffectSettings access -- which is no-op -- and to the side, an EffectSettingsAccess connected to a temporary RealtimEffectState which we don't need anyway.

Reply:
However, I may want to redraft this anyway.  It may be that one `Flush()` in an idle time handler of the `EffectUIHost` is all we need.